### PR TITLE
Try using VS Code API for Emmet tests

### PR DESCRIPTION
**Pull request by rzhao271**
Ref https://github.com/microsoft/vscode/issues/162295

This PR is experimental. I'm not sure whether it can resolve the fact that the Emmet integration tests sometimes hang. The PR adds some assertions, and also changes the filesystem calls to use the VS Code API, considering that the file is already using that API in the first place.

### CODE REVIEW

**Review by Kaparrs — DISMISSED** [review]

**Review by sumon1991 — DISMISSED** [review]

File: extensions/emmet/src/test/testUtils.ts [hunk 1]
```diff
@@ -17,17 +17,11 @@ export function rndName() {
...
-			}
-
-			resolve(vscode.Uri.file(tmpFile));
-		});
-	});
+export async function createRandomFile(contents = '', fileExtension = 'txt'): Promise<vscode.Uri> {
+	const tmpFile = join(os.tmpdir(), rndName() + '.' + fileExtension);
+	const fileUri = vscode.Uri.file(tmpFile);
+	await vscode.workspace.fs.writeFile(fileUri, Buffer.from(contents));
+	return fileUri;
```
Comment (jrieken) [inline extensions/emmet/src/test/testUtils.ts:24]: Using the FS API is always good but I doubt that will help to make tests more stable. 

For the API tests themselves we went away from tmp-file and OS dependencies. That reduces the dependencies of your tests. For the API tests stability was improved. 

This is how/where things are setup https://github.com/microsoft/vscode/blob/36fdf6b697cba431beb6e391b5a8c5f3606975a1/extensions/vscode-api-tests/src/utils.ts#L10-L17

Comment (jrieken) [reply extensions/emmet/src/test/testUtils.ts:24]: There is unfortunately no code reuse so you'll need to copy-paste this and the memfs-file

**Review by jrieken — DISMISSED** [review]

**Comment by jrieken** [comment]
change looks good but tests are failing...

**Comment by rzhao271** [comment]
I have reverted the TestFS changes for now

**Pull request closed without merging**